packages/wp-dashboard/src/components/editorSettings/shopping/shopify/index.js

```diff
@@ -150,6 +182,35 @@ function ShopifySettings({
     [onSaveAccessToken]
   );
 
+  const onTestConnection = useCallback(async () => {
+    setHasConnectionError(false);
+    setIsTestingConnection(true);
+    speak(TEXT.CONNECTION_CHECKING);
+    try {
+      await getProducts();
+      setTestConnectionStatus(TEXT.CONNECTION_SUCCESS);
+      speak(TEXT.CONNECTION_SUCCESS);
+    } catch (e) {
+      setHasConnectionError(true);
+      setTestConnectionStatus(TEXT.CONNECTION_ERROR_DEFAULT);
+      speak(TEXT.CONNECTION_ERROR_DEFAULT);
+    } finally {
+      setIsTestingConnection(false);
+    }
+  }, [getProducts, speak]);
+
+  const canTestConnection = useCallback(() => {
+    const hasHost = hostInputError === '' && host !== '';
+    const hasToken = accessTokenInputError === '' && accessToken !== '';
+    return hasHost && hasToken && !isTestingConnection ? true : false;
+  }, [
+    host,
+    accessToken,
+    accessTokenInputError,
+    hostInputError,
+    isTestingConnection,
+  ]);
```

Comment:
This does not need to be a callback, since it's just called directly anyway. `useMemo` would be more accurate, but also overkill. It's just a simple `const`.

I'd simplify to something like

```js
const canTestConnection = host && !hostInputError && accessToken && !accessTokenInputError && !isTestingConnection;
```